### ISSUE TITLE
Courses by subject not listed in alphabetical order

### ISSUE THREAD

**Issue by joycelynmng** [hotseatio/hotseat.io#57]
Currently, when you search for a subject and not a certain course it is hard to do because it is really messy and not numerically ordered it is random. 
![Screenshot 2023-02-08 at 8 56 17 PM](https://user-images.githubusercontent.com/95897133/217721809-6225c5c4-74ab-439f-abf3-527c8add6fac.png)
![Screenshot 2023-02-08 at 8 56 27 PM](https://user-images.githubusercontent.com/95897133/217721824-cd335240-b410-4ad7-9e8e-ae3d41717681.png)

**Comment by TimothyGu** [comment]
This is not easily fixable. The second screenshot is the search results for the query "physics", where numerical order may not be the best order. Suppose if the COM SCI subject area has a class called "Physics in computer graphics", it is unclear where to slot that CS class relative to the physics classes in the search results.

**Comment by joycelynmng** [comment]
Would it be possible to change it so that in the second slide when you are searching for just course subject names such as "physics" the course subject area pops up so that it is already in numerical order? When people search up a key term like "physics" or "math" its usually to look up the whole subject area as they are looking to browse through the course offerings within that subject area. This way people will not have to go to the registrar to look up each course code and then come back to hotseat. Additionally they would not have to navigate back to the home page to select the course subject area if they are looking for a different subject.

**Comment by TimothyGu** [comment]
Yeah, I think that'd be a good idea. We could have the search page return subject areas as well (much like how we already have instructor and courses show up on the same page), and make sure the subject areas display on the page before courses. That seems to be more doable.

Do you want to take a stab at it? Nathan probably knows how this works better, but I believe this requires adding a few lines like these to `subject_area.rb`:

https://github.com/hotseatio/hotseat.io/blob/8dd1086d6cc951e493af4b548d1afa3779f94ba1/app/models/course.rb#L6
https://github.com/hotseatio/hotseat.io/blob/8dd1086d6cc951e493af4b548d1afa3779f94ba1/app/models/course.rb#L77-L83

These lines would add subject areas to the search index, and also dictate how the subject area is to be displayed on the search page (e.g., "COM SCI: Computer Science").
